### ISSUE TITLE
feature warn when using 'exit' as last command in a subshell...

### ISSUE THREAD

**Issue by nihilus** [koalaman/shellcheck#1064]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
foo
([ -f "bar" ] && fubar; exit)
foobar

```

#### Here's what I wanted or expected to see:

Warning: exit will only exit the subshell with exitcode 0.

**Comment by Nightfirecat** [comment]
Perhaps this could be covered alongside #1031? Seems to me they should both be covered by whatever implementation is put in place.

**Comment by jleffler** [comment]
If the file `bar` does not exist, the subshell exits with status 1 (because the test fails and the `fubar` is not executed.  So, the proposed warning message is wrong.  You'd get the same exit status from the sub-shell even if the exit were omitted — but that's different from "will only exit with status 0".

**Comment by koalaman** [comment]
As commented, it's superfluous but not wrong. There are definitely cases when `exit` doesn't have the intended scope, but this doesn't seem to be a common one.
